commit 51907f07f9022ad617e3ed8c3869bcfca8508f4d (refs/changes/27/2627/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-09-20T11:58:25-07:00 (2 years, 1 month ago)
    
    TOOLS-1842 'sdcadm create SERVICE-NAME --server=UUID' fails with 'Unknown change type: instances'

diff --git a/CHANGES.md b/CHANGES.md
index 3e52fa2..aac37c5 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,10 @@
 
 # sdcadm Changelog
 
+## 1.16.1
+
+- TOOLS-1842 Fix 'sdcadm create SERVICE-NAME -s SERVER' which was failing
+  after TOOLS-1770 changes.
 
 ## 1.16.0
 
diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index f492199..44cdbeb 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -2776,14 +2776,13 @@ SdcAdm.prototype.genUpdatePlan = function genUpdatePlan(options, cb) {
 
             var changeFromSvc = {};
             var changeFromInst = {};
-            var i, ch, typeTarg, svc;
+            var i, ch, svc;
             for (i = 0; i < changes.length; i++) {
                 ch = changes[i];
-                // e.g. 'update-service' -> 'service'
-                // (Some instance changes like 'delete' or 'create' do not
-                // include the two pieces).
-                typeTarg = ch.type.split('-')[1] || 'instance';
-                if (typeTarg === 'service') {
+                switch (ch.type) {
+                case 'delete-service':
+                case 'update-service':
+                case 'create-instances':
                     svc = ch.service.name;
                     if (changeFromSvc[svc]) {
                         next(new errors.UpdateError(format(
@@ -2793,7 +2792,14 @@ SdcAdm.prototype.genUpdatePlan = function genUpdatePlan(options, cb) {
                         return;
                     }
                     changeFromSvc[svc] = ch;
-                } else if (typeTarg === 'instance') {
+                    break;
+                case 'delete':
+                    // `validateChanges` is too lenient that it allows "delete".
+                    // Eventually it should become more strict to require
+                    // explicit "delete-instance".
+                    // fall through
+                case 'delete-instance':
+                case 'update-instance':
                     var inst = (ch.instance) ? ch.instance.instance : null;
                     if (changeFromInst[inst]) {
                         next(new errors.UpdateError(format(
@@ -2803,16 +2809,17 @@ SdcAdm.prototype.genUpdatePlan = function genUpdatePlan(options, cb) {
                         return;
                     }
                     changeFromInst[inst] = ch;
-                } else {
+                    break;
+                default:
                     next(new errors.UpdateError(format(
-                        'Unknown change type: %s', typeTarg)));
+                        'Unknown change type: %s', ch.type)));
                     return;
                 }
             }
             for (i = 0; i < changes.length; i++) {
                 ch = changes[i];
-                typeTarg = ch.type.split('-')[1] || 'instance';
-                if (typeTarg === 'instance') {
+                if (ch.type === 'update-instance' ||
+                    ch.type === 'delete-instance') {
                     svc = ch.service.name;
                     if (changeFromSvc[svc]) {
                         next(new errors.UpdateError(format(
diff --git a/package.json b/package.json
index f91ef8e..30d0a29 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.16.0",
+  "version": "1.16.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
